data/src/history.rs

```diff
@@ -253,16 +354,34 @@ impl History {
                 server,
                 kind,
                 messages,
+                metadata,
                 ..
-            } => append(&server, &kind, messages).await,
+            } => append(&server, &kind, messages, &metadata).await,
             History::Full {
                 server,
                 kind,
                 messages,
+                metadata,
                 ..
-            } => overwrite(&server, &kind, &messages).await,
```

Comment:
I need to call metadata.updated here to ensure we save updates metadata when we close